repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/classloader/ClassLoaderTest.groovy

```diff
@@ -92,7 +82,8 @@ class ClassLoaderTest extends ReposeValveTest {
         repose.start(killOthersBeforeStarting: false,
                 waitOnJmxAfterStarting: false)
 
-        repose.waitForNon500FromUrl(url)
+        reposeLogSearch.awaitByString("Repose ready", 1, 30)
```

Comment:
Also that method is a hot mess, i very much prefer different methods that do different ready checks to one that has all sorts of weird maybe behavior.